server/app/tests/test_registry.py

```diff
@@ -86,73 +117,124 @@ def test_handle_registry_request_no_push_access(registry_service: RegistryServic
     registry_service.repo_service.find_by_canonical_name.return_value = MagicMock(id=1, canonical_name="test/repo")
     registry_service.access_control_service.has_write_access.return_value = False
 
-    scope = "repository:test/repo:push"
+    scopes = ["repository:test/repo:push"]
     username = "testuser"
     password = "password"
     service = "docker-registry"
 
     with pytest.raises(HTTPException) as excinfo:
-        registry_service.handle_registry_request(username, password, scope, service)
+        registry_service.handle_registry_request(username, password, scopes, service)
 
     assert excinfo.value.status_code == 401
     assert "User testuser cannot push to repo test/repo" in str(excinfo.value.detail)
 
+def test_handle_registry_request_push_with_mount_no_pull_access(registry_service: RegistryService, mock_jwt_encode):
+    registry_service.user_service.exists_with_credentials.return_value = True
+    registry_service.user_service.find_by_username.return_value = MagicMock(id=1, username="testuser")
+    registry_service.access_control_service.has_write_access.return_value = True
+
+    def check_read_access(user_id, repo_id):
+        if (user_id, repo_id) == (1, 1):
+            return False
+        elif (user_id, repo_id) == (1, 2):
+            return True
+        return None
+
+    registry_service.access_control_service.has_read_access.side_effect = check_read_access
+
+    def find_repo_by_name(name):
+        if name == "test/repo1":
+            return MagicMock(id=1, canonical_name="test/repo1")
+        elif name == "test/repo2":
+            return MagicMock(id=2, canonical_name="test/repo2")
+        return None
+
+    registry_service.repo_service.find_by_canonical_name.side_effect = find_repo_by_name
+
+    scopes = ["repository:test/repo2:pull", "repository:test/repo2:pull,push", "repository:test/repo1:pull"]
+    username = "testuser"
+    password = "password"
+    service = "docker-registry"
+
+    with pytest.raises(HTTPException) as excinfo:
+        registry_service.handle_registry_request(username, password, scopes, service)
+    
+    assert excinfo.value.status_code == 401
+    assert "User testuser cannot pull from repo test/repo1" in str(excinfo.value.detail)
+
```

Comment:
Though, I should point out that testing exception messages like this can cause issues if we ever change the exception message. A better approach is to have some structured data, but I don't know how you would do this except for creating another exception class that inherits from `HTTPException`. Seems like a waste of time, in my opinion.